### PR TITLE
SNAP-902 Display Phone Numbers In Order By Type

### PR DESCRIPTION
**Pull request by ggomez-cambriasolutions**
### Jira Story

https://osi-cwds.atlassian.net/browse/SNAP-902

## Description
Display Phone numbers in a designated order by type on the Snapshot Person Card. 

1. Home
2. Cell
3. Other
4. Work

## Tests
- [x] I have included unit tests 
- [ ] I have included feature tests 
- [ ] I have included other tests 
- [ ] I have NOT included tests 

## Types of changes

- [ ] New feature (non-breaking change which adds functionality)
- [x] Refactoring (No behavioral changes)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [x] I have ran all tests for the project.
- [x] I have ran lint/rubocop check for the changed/new files.
- [x] My code is in a stable state ready to be deployable, but not necessarily complete.
- [x] I promise on my honor as a CWDS developer to ensure I don't break the build, to check Lint/Rubocop, use best practices, to leave the code in better shape than I found it, and to have a good day.

### CODE REVIEW

**Comment by ggomez-cambriasolutions** [comment]
jenkins retest this please

**Comment by ggomez-cambriasolutions** [comment]
jenkins retest this please

**Comment by ggomez-cambriasolutions** [comment]
jenkins retest this please

**Comment by pokharelmanish** [comment]
@ggomez-cambriasolutions  I am seeing other first and then home, Is this the expected behavior? 
screen shot 2019-02-05 at 1 04 35 pm

**Comment by ggomez-cambriasolutions** [comment]
Just so everyone is aware, the screen shot above has 3 addresses, each with 1 phone number. The sorting algorithm sorts phone numbers by type, for each address. It is not intended to do a final sort of all the phone numbers, for all addresses. In this example above, the sorting is correct. It would be incorrect if all the phone numbers in the screen shot were for 1 address.

**Comment by ggomez-cambriasolutions** [comment]
jenkins retest this with the magic touch

File: spec/javascripts/utils/sortPhoneNumberTypesSpec.js [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+  ],
+]
+
+describe('Sort Phone Numbers by Type', () => {
+  let sortedPhoneNumbers
+  beforeEach(() => {
+    sortedPhoneNumbers = sortPhoneNumberTypes(phoneNumbers)
+  })
+
+  it('sorts the Home type first', () => {
```
Comment (mlipardo) [inline spec/javascripts/utils/sortPhoneNumberTypesSpec.js]: I think this test should be an edge case like if its an empty array or there is no type in the phone number since there is already a test - _sort all numbers in the correct_ ,

**Comment by ggomez-cambriasolutions** [comment]
jenkins retest this please with the magic touch

**Comment by ggomez-cambriasolutions** [comment]
jenkins retest this please, with the super duper magic touch this time

**Comment by ggomez-cambriasolutions** [comment]
jenkins retest this please

File: spec/javascripts/utils/sortPhoneNumberTypesSpec.js (hunk 1 above)
Comment (ggomez-cambriasolutions) [reply spec/javascripts/utils/sortPhoneNumberTypesSpec.js]: Thanks for the review. I've added a few test cases that cover phone number edge cases.

File: app/javascript/data/address.js [hunk 2]
```diff
@@ -5,47 +5,77 @@ import {getZIPErrors} from 'utils/zipValidator'
 
 export const AddressPropType = PropTypes.shape({
   city: PropTypes.string,
+  phone_numbers: PropTypes.arrayOf(
```
[the remark this answers is not in the document]
Comment (ggomez-cambriasolutions) [reply app/javascript/data/address.js]: This prop type is not needed, so I just removed it. The components that are using this prop type are not rendering or dealing with phone numbers.

**Comment by ggomez-cambriasolutions** [comment]
jenkins retest this please

**Comment by ggomez-cambriasolutions** [comment]
jenkins retest this please

**Comment by pokharelmanish** [comment]
retest this please

**Comment by ggomez-cambriasolutions** [comment]
jenkins retest this please
